dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EventDAO.java

```diff
@@ -107,8 +128,9 @@ public List<Event> findAllByOrganizationIdsOrEntryIds(Set<Long> organizationIds,
         query.select(event);
 
         int primitiveOffset = MoreObjects.firstNonNull(offset, 0);
-        TypedQuery<Event> typedQuery = currentSession().createQuery(query).setFirstResult(primitiveOffset).setMaxResults(newLimit);
-        return typedQuery.getResultList();
+        TypedQuery<Event> typedQuery = currentSession().createQuery(query);
+        List<Event> filteredEvents = filterCategoryEvents(loggedInUser, typedQuery.getResultList());
+        return filteredEvents.subList(primitiveOffset, Math.min(primitiveOffset + newLimit, filteredEvents.size()));
```

Comment:
Nit: I see these two lines (filter & sublist) 4 times in the file; you could combine them into one method.